rust/agama-lib/src/base_http_client.rs

```diff
@@ -43,12 +43,19 @@ impl BaseHTTPClient {
     /// Uses `localhost`, authenticates with [`AuthToken`].
     pub fn new() -> Result<Self, ServiceError> {
         Ok(Self {
-            client: Self::authenticated_reqwest_client()?,
+            client: Self::authenticated_client()?,
             ..Default::default()
         })
     }
 
-    fn authenticated_reqwest_client() -> Result<reqwest::Client, ServiceError> {
+    pub fn new_unauthenticated_with_url(url: String) -> Result<Self, ServiceError> {
```

Comment:
```base_url``` is public, so no need to initialize that via constructor. Moreover it adds some overhead when you want unauthorized client but with a default API url (which for now might be the most expected case in my POV)

Reply:
You're right, it is not needed